### DISCUSSION TITLE
Origin for Actions

### DISCUSSION THREAD

**Discussion by aetkie** [UNC-Project-Open-AAC/OS-DPI#227] [Q&A] [answered]
Is the origin for writing an action the name of the component or the component itself?

For instance, if the action originates in the grid do I write **grid** or since the grid is named "core" do I write **core**?

<img width="524" alt="Screenshot 2023-10-24 at 9 54 21 AM" src="https://github.com/UNC-Project-Open-AAC/OS-DPI/assets/113473933/8758c74a-41d8-46c2-afd2-32baf61074dc">

**Comment by gbishop** [comment]
It is the name of the component in most cases. In the case of Emit from a
Method it is the name of the Method.

**Comment by sofia-bg** [comment, marked as answer]
I think this is a good reason in examples to NOT name the component what the component is. So in examples avoiding calling the grid grid and doing what you have done and naming it core. 

It is why the display window does nothing until you give it a name, then you can prescribe actions to it using whatever you named it (which of course I tend to name it display...)

**Reply by aetkie** [reply]
Yes, helpful! Thanks

**Comment by gbishop** [comment]
I'm thinking we should change the default names. Right now the default name
for the Grid is grid simply because I didn't think about it. Maybe the
default name should be "a grid"? or "Grid 17" where the number is different
for each grid?

**Comment by aetkie** [comment]
I like the number idea. For a design I was working on with multiple grids I started naming them "grid first", "grid second" etc because I knew they needed to stay in a certain order and when collapsed, I was losing sight of the order if calling them "grid core", "grid emotions" etc.
